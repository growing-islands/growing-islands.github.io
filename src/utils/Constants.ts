type BreakpointsType = {
  PC: number
  SP: number
}

export const BREAKPOINTS: BreakpointsType = {
  PC: 769,
  SP: 768,
}

export const PAGE_INFO = {
  URL: process.env.NEXT_PUBLIC_SITE_URL || 'https://',
  TWITTER: 'https://twitter.com/yumelab1214',
  FACEBOOK: 'https://www.facebook.com/yumelab.co.jp/',
}

export const INTERSECTION_OPTIONS = {
  root: null,
  rootMargin: '0px 0px -100%',
  threshold: 0,
}
