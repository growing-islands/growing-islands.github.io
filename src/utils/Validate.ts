/**
 * Form Validate関連の変数
 */

export const ERROR_MESSAGES = {
  required: 'この項目は入力必須です',
  maxLength: '入力できる文字数を超えています',
  mixLength: '文字数が足りません',
  max: '数値が大きすぎます',
  mix: '数値が小さすぎます',
  num: '半角数字で入力して下さい',
  hiragana: 'ひらがなで入力してください',
  fullKana: '全角カタカナで入力してください',
  tel: '電話番号が正しい形式ではありません',
  email: 'メールアドレスの形式が正しくありません',
  password:
    '英文字（大文字、小文字）、数字をそれぞれ最低一文字以上入れた8〜16文字で入力してください',
  dependent_char: '使用できない文字列が含まれています',
  url: '「http://」もしくは「https://」から始まるURLを入力してください',
}

// NOTE： 正規表現を記述するためにESLintを無効
/*eslint-disable no-useless-escape */
/*eslint-disable no-irregular-whitespace */
export const REGEX = {
  num: /^[0-9]+$/,
  alphanumeric: /^[0-9a-zA-Z]+$/,
  hiragana: /^[ぁ-んー　]*$/,
  fullKana: /^[ァ-ヶー　]*$/,
  halfKana: /^[ｦ-ﾟ]*$/,
  tel: /^(0{1}\d{9,10})$/,
  email:
    /^([\w!#$%&'*+\-\/=?^`{|}~]+(\.[\w!#$%&'*+\-\/=?^`{|}~]+)*|"([\w!#$%&'*+\-\/=?^`{|}~. ()<>\[\]:;@,]|\\[\\"])+")@(([a-zA-Z\d\-]+\.)+[a-zA-Z]+|\[(\d{1,3}(\.\d{1,3}){3}|IPv6:[\da-fA-F]{0,4}(:[\da-fA-F]{0,4}){1,5}(:\d{1,3}(\.\d{1,3}){3}|(:[\da-fA-F]{0,4}){0,2}))\])$/,
  password:
    /^(?=.*?[a-z])(?=.*?[A-Z])(?=.*?\d)[a-zA-Z0-9!"#$%&'()\*\+\-\.,\/:;<=>?@\[\\\]^_`{|}~]{8,16}$/,
  dependent_char:
    /^(?!.*(\u2460|\u2461|\u2462|\u2463|\u2464|\u2465|\u2466|\u2467|\u2468|\u2469|\u246a|\u246b|\u246c|\u246d|\u246f|\u2470|\u2471|\u2472|\u2473|\u2160|\u2161|\u2162|\u2163|\u2164|\u2165|\u2166|\u2167|\u2168|\u2169|\u3349|\u3314|\u3322|\u334d|\u3318|\u3327|\u3303|\u3336|\u3351|\u3357|\u330d|\u3326|\u3323|\u332b|\u334a|\u333b|\u339c|\u339d|\u339e|\u338e|\u338f|\u33c4|\u33a1|\u337b|\u301d|\u301f|\u2116|\u33cd|\u2121|\u32a4|\u32a5|\u32a6|\u32a7|\u32a8|\u3231|\u3232|\u3239|\u337e|\u337d|\u337c|\u222e|\u221f|\u22bf|\u2170|\u2171|\u2172|\u2173|\u2174|\u2175|\u2176|\u2177|\u2178|\u2179|\uffe2|\uffe4|[\uD800-\uDBFF][\uDC00-\uDFFF])).*$/,
  url: /^(https?)(:\/\/[-_.!~*\'()a-zA-Z0-9;\/?:\@&=+\$,%#]+)$/,
}
/*eslint-enable no-useless-escape */
/*eslint-enable no-irregular-whitespace */
