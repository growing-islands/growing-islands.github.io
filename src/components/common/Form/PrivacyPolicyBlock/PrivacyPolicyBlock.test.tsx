import { render } from '~tests/libs/utils'
import { PrivacyPolicyBlock } from '.'

describe('components/common/Form/PrivacyPolicyBlock', () => {
  it('Snapshot', () => {
    const { asFragment } = render(<PrivacyPolicyBlock />)

    expect(asFragment()).toMatchSnapshot()
  })
})
