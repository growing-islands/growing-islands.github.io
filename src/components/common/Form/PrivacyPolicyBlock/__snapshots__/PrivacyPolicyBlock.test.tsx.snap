// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/common/Form/PrivacyPolicyBlock Snapshot 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <div>
      <p>
        プライバシーポリシー
        <br />
        株式会社ユメラボ（以下「当社」）では、個人情報に関する法令およびその他の規範を遵守し、お客様の大切な個人情報の保護に万全を尽くします。
      </p>
      <p>
        1 個人情報の収集について
        <br />
        当社では、次のような場合に必要な範囲で個人情報を収集することがあります。
      </p>
      <p>
        ・当社へのお問い合わせ時
        <br />
        ・当社へのサービスお申し込み時
      </p>
      <p>
        2 個人情報の利用目的について
        <br />
        当社は、お客様から収集した個人情報を次の目的で利用いたします。また、目的の範囲内であっても、違法または不当な行為を助長または誘発するおそれがある方法でお客様の個人情報を利用しません。
      </p>
      <p>
        ・お客様への連絡のため
        <br />
        ・お客様からのお問い合せに対する回答のため
        <br />
        ・お客様へのサービス提供のため
      </p>
      <p>
        3 個人情報の第三者への提供について
        <br />
        当社では、お客様より取得した個人情報を第三者に開示または提供することはありません。
        <br />
        ただし、次の場合は除きます。
      </p>
      <p>
        ・ご本人の同意がある場合
        <br />
        ・警察からの要請など、官公署からの要請の場合
        <br />
        ・法律の適用を受ける場合
      </p>
      <p>
        4 個人情報の開示、訂正等について
        <br />
        当社は、お客様ご本人からの自己情報の開示、訂正、削除等のお求めがあった場合は、確実に応じます。
      </p>
      <p>
        5 苦情や相談への対応と安全管理措置について
        <br />
        当社は、個人情報の取扱いに関するお客様からの苦情や相談を受け付け、適切かつ迅速に対応いたします。また、お客様からの開示・訂正・追加・削除、利用又は提供の停止などのご要望に対しても、迅速適切に対応いたします。これらの受付窓口は、8に記載の窓口となります。
        <br />
        また、当社は、個人情報保護法に基づき、個人情報について、漏えい、滅失又は毀損の防止等、その他の安全管理のために必要かつ適切な安全管理措置を講じます。保有個人データの安全管理のために講じた措置について、8に記載の窓口に問い合わせいただいた場合、当社は個人情報保護法の定めに従い、お客様ご本人の求めに応じて遅滞なく回答を行います。
      </p>
      <p>
        6 プライバシーポリシーの変更
        <br />
        本ポリシー内容は、ユーザーに通知することなく、変更することができるものとします。また、当社が別途定める場合を除いて、変更後のプライバシーポリシーは、本ウェブサイトに掲載したときから効力を生じるものとします。
      </p>
      <p>
        7 当社名称及び代表者氏名
        <br />
        会社名：株式会社ユメラボ
        <br />
        代表取締役：深松一平
        <br />
        住所：千葉県浦安市北栄1-1-3 柳内ビル 3階
      </p>
      <p>
        8 問い合わせ窓口
        <br />
        本ポリシーに関するお問い合わせは、本ページのお問い合わせフォームからお願いいたします。
      </p>
    </div>
  </div>
</DocumentFragment>
`;
